server/lib/settings/migrator.ts

```diff
@@ -10,19 +10,42 @@ export const runMigrations = async (
   settings: AllSettings,
   SETTINGS_PATH: string
 ): Promise<AllSettings> => {
-  const migrations = fs
-    .readdirSync(migrationsDir)
-    .filter((file) => file.endsWith('.js') || file.endsWith('.ts'))
-    // eslint-disable-next-line @typescript-eslint/no-var-requires
-    .map((file) => require(path.join(migrationsDir, file)).default);
-
   let migrated = settings;
 
   try {
+    // we read old backup and create a backup of currents settings
+    const BACKUP_PATH = SETTINGS_PATH.replace('.json', '.old.json');
+    const oldBackup = fs.existsSync(BACKUP_PATH)
+      ? fs.readFileSync(BACKUP_PATH)
+      : null;
+    fs.writeFileSync(BACKUP_PATH, JSON.stringify(settings, undefined, ' '));
+
+    const migrations = fs
+      .readdirSync(migrationsDir)
+      .filter((file) => file.endsWith('.js') || file.endsWith('.ts'));
+
     const settingsBefore = JSON.stringify(migrated);
 
     for (const migration of migrations) {
-      migrated = await migration(migrated);
+      try {
+        logger.debug(`Checking migration '${migration}'...`, {
+          label: 'Settings Migrator',
+        });
+        // eslint-disable-next-line @typescript-eslint/no-var-requires
+        const fn = require(path.join(migrationsDir, migration)).default;
+        const newSettings = await fn(migrated);
+        if (JSON.stringify(migrated) !== JSON.stringify(migrated)) {
```

Comment:
shouldn't this be:
```suggestion
if (JSON.stringify(migrated) !== JSON.stringify(newSettings)) {
```
I guess I am a bit confused as to why you're comparing it with itself

Reply:
Yep idk how it end up like this